agent/config.m4

```diff
@@ -117,8 +117,20 @@ if test "$PHP_NEWRELIC" = "yes"; then
   dnl Check for axiom.
   dnl The -ldl in the first argument is there to get the library check to pass.
   dnl Without it, we get an undefined reference to dladdr as the dl lib must
-  dnl come after the axiom lib when linking.
-  PHP_CHECK_LIBRARY(axiom -ldl, nro_new, [
+  dnl come after the axiom lib when linking. Note: this does not apply to
+  dnl FreeBSD builds, and if present, causes errors when building the agent.
+  dnl If other platforms are known to have the same issue, simply change the
+  dnl   *freebsd*)
+  dnl line to something like
+  dnl   *freebsd*|*darwin*)
+  dnl See the other `case $host in` block later in this file for known targets
+  case "$host" in
+    *freebsd*) AXIOM_CHECK="axiom"
+               ;;
+    *) AXIOM_CHECK="axiom -ldl"
+               ;;
+  esac
```

Comment:
The definition of `AXIOM_CHECK` could go in the FreeBSD clause of the case statement back on line 84. That would reduce the number of case statements in the file and put all platform specific definitions in one place.

Reply:
I kept your logic where it was, because I think it is most clear there, but moved only FreeBSD's logic to the system-dependent section with a comment in the original location.

I thought setting the var in each system obscured FreeBSD's quirkiness of being the outlier here, so opted to emphasize that it stands out a little more.  Let me know if that sounds good to you